pilot/pkg/config/kube/gateway/backend_policies.go

```diff
@@ -175,7 +175,10 @@ func DestinationRuleCollection(
 					rbSet = true
 					spec.TrafficPolicy.RetryBudget = pol.RetryBudget
 				}
-				parents = append(parents, pol.Source.Kind.String()+"/"+pol.Source.Namespace+"."+pol.Source.Name)
+				parentName := pol.Source.Kind.String() + "/" + pol.Source.Namespace + "." + pol.Source.Name
+				if !slices.Contains(parents, parentName) {
+					parents = append(parents, pol.Source.Kind.String()+"/"+pol.Source.Namespace+"."+pol.Source.Name)
```

Comment:
```suggestion
					parents = append(parents, parentName)
```

nit

Reply:
I'm sorry, I don't know how I could have missed that...

Done.